### PR TITLE
fix: Redirect user to edit url after a successful page creation

### PR DESCRIPTION
**Pull request by sakhawy**
## Description
Added the wizard property ``edit_mode_on_success`` and defaulted it to True to redirect newly created pages to edit mode instead of preview mode. The related documentation was updated.

## Related resources

* #7688 

## Checklist

* [X] I have opened this pull request against ``develop-4``
* [X] I have added or modified the tests when changing logic
* [X] I have followed [the conventional commits guidelines](https://www.conventionalcommits.org/) to add meaningful information into the changelog
* [x] I have read the [contribution guidelines ](https://github.com/django-cms/django-cms/blob/develop/CONTRIBUTING.rst) and I have joined #workgroup-pr-review on [Slack](https://www.django-cms.org/slack) to find a “pr review buddy” who is going to review my pull request.

### CODE REVIEW

**Comment by fsbraun** [comment]
Hey @sakhawy ! This looks good to me! Just one thing: Should we also test that we get the preview URL if `edit_mode_on_success` is false?

**Comment by sakhawy** [comment]
@fsbraun Thanks! Done.

**Review by fsbraun — APPROVED** [review]
Well done! Looks good!

**Review by marksweb — CHANGES_REQUESTED** [review]
Just a minor point.

File: cms/tests/test_wizards.py [hunk 1]
```diff
@@ -142,6 +146,53 @@ def test_get_success_url(self):
...
+
+        with patch.object(extension, 'toolbar_enabled_models', {Page: page}):
+            url = self.page_wizard.get_success_url(
+                page, language="en")
+            self.assertEqual(
+                url, get_object_edit_url(page, language="en"))
+
+    def test_get_preview_url(self):
+        wizard_preview_mode = Wizard(
+            title=_(u"Page"),
```
[the remark this answers is not in the document]
Comment (sakhawy) [reply cms/tests/test_wizards.py]: Thanks!

**Review by marksweb — APPROVED** [review]

**Pull request merged**